### PR TITLE
Update custom_platform_hooks.md with dir info

### PR DESCRIPTION
**Pull request by travisreed-wf**
*Description of changes:*
Add a specific reference to the directory location
Include an example

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by dankhen** [comment]
Hi @travisreed-wf ,
Thanks for this submission! Can you clarify something for me?
`.ebextensions` is a local directory in your sources. It looks like your command tries to copy a file from an `.ebextensions` directory on your instances to the hooks directory. Does this work? I would expect including the content of the hooks script as a file in the `.ebextensions` file.

**Comment by travisreed-wf** [comment]
@dankhen `.ebextensions` is a local directory, but as far as I can tell, aws makes it available to the instance during the environment update.

This is a pattern that we found on in the [datadog documentation](https://www.datadoghq.com/blog/deploy-datadog-aws-elastic-beanstalk/) and have been using for our own setup.

**Comment by travisreed-wf** [comment]
@dankhen what would you like me to do with this PR to move forward?

**Comment by dankhen** [comment]
Please wait for just a bit longer. I am finding out with our engineers. Sorry that it's taking a while.

**Comment by dankhen** [comment]
Thanks for your patience @travisreed-wf . So here's where we stand - I'll try to give you a short answer, then all the details.
### TL;DR:
We'd like to keep the first change in your PR (the one where you specify the parent hook directory). We'd like not to add the ebextensions example. If you can remove that second change, I'll be happy to merge the PR.

### Details
1. Hooks are currently documented as a custom platform feature. The recommended way to put hook scripts on a custom platform is to bake them into the AMI, not to place them with .ebextensions. The reason is that with .ebextensions, depending on which hooks you edit, they may not be placed on the instance in time for the hook to execute.
I will clarify this fact in the topic, as it's missing right now.
1. FYI for completeness, even if we wanted to go the .ebextensions way: your `cp` command assumes that the entire directory tree exists - `/opt/elasticbeanstalk/hooks/appdeploy/pre/`. On a custom platform, this isn't necessarily the case. You can see that in the datadog example you link to, they're doing some checks and `mkdir` commands prior to copying; and even that example doesn't seem to be complete, because they're only testing for the leaf directories, but some of the parents might not exist as well on a custom platform.
1. A simpler approach would be to embed the content of the file in the .ebextensions file, using the `files:` key. This is what we do in most of our examples. Typically, you use `files:` to place a single copy of a file on the instance, and you then use `cp` commands to copy it to additional locations if necessary (which isn't the case in your example). The advantage of using `files:` is that Elastic Beanstalk takes care of creating any components of the directory tree you specify, in case they're missing.

Again, if you modify your PR to leave only the first change, I'll be happy to merge it.

Thanks!

**Comment by travisreed-wf** [comment]
@dankhen thanks for the info~ I have updated this PR.
